velox/expression/Expr.cpp

```diff
@@ -1200,7 +1234,7 @@ void Expr::evalAll(
   }
   bool tryPeelArgs = deterministic_ ? true : false;
   bool defaultNulls = vectorFunction_->isDefaultNullBehavior();
-
+  bool throwArgumentErrors = context.throwOnError() && !defaultNulls;
```

Comment:
flat-no-nulls fast path is enabled only when context.throwOnError is true. This change may inadvertently disable flat-no-nulls fast path. Can we make sure this doesn't happen?

In flat-no-nulls fast path we know that there are no nulls in the input and no function can produce a null on non-null input. Hence, we know it is not possible for any error to be masked by a null. Hence, we can disable this new logic.